File: src/main/java/iuh/fit/backend/model/Address.java
package iuh.fit.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Entity
@Table(name = "address")
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String receiverName;

    private String receiverPhone;

    private String province;

    private String district;

    private String ward;

    private String specifics;

    private int main;

    @ManyToOne
    @JoinColumn(name = "customer_id", nullable = false)
    @JsonIgnore
    @ToString.Exclude
    private Customer customer;
}

File: src/types/Address.ts
// Address type matching backend model
export interface Address {
    id?: number
    receiverName: string
    receiverPhone: string
    main: number
    province: string
    district: string
    ward: string
    specifics: string
    isDefault: boolean
}

export interface AddressFormData {
    receiverName: string
    receiverPhone: string
    main: string
    province: string
    district: string
    ward: string
    specifics?: string
    isDefault?: boolean
}
